modules/artifact/src/main/java/org/mule/module/artifact/classloader/soft/buster/ComposedSoftReferenceBuster.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.artifact.classloader.soft.buster;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.WeakHashMap;
+
+public class ComposedSoftReferenceBuster implements SoftReferenceBuster {
+
+  private static final ComposedSoftReferenceBuster INSTANCE = new ComposedSoftReferenceBuster();
+
+  private static WeakHashMap<ClassLoader, List<SoftReferenceBuster>> busters =
```

Comment:
Could this logic be added to `DefaultResourceReleaser` instead of having a singleton with a map at the runtime level?

Reply:
Yes, I have moved this to the DefaultResourceReleaser. No need for the soft reference busters hierarchy.